### PR TITLE
Add 'Button' component & update to current best practise

### PR DESCRIPTION
**Pull request by emrysal**

### CODE REVIEW

File: components/dialog/ConfirmationDialogContent.tsx [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-  confirmBtnText,
-  cancelBtnText,
-  onConfirm,
-  children,
-}) {
-  confirmBtnText = confirmBtnText || "Confirm";
-  cancelBtnText = cancelBtnText || "Cancel";
+export type ConfirmationDialogContentProps = {
+  confirmBtnText?: string;
+  cancelBtnText?: string;
```
Comment (KATT) [inline components/dialog/ConfirmationDialogContent.tsx:9]: I can't see that you have any defaults - either add fallbacks or make them mandatory


```suggestion
  confirmBtnText: string;
  cancelBtnText: string;
```

File: components/dialog/ConfirmationDialogContent.tsx [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+  onConfirm: (event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
+  title: string;
+  alert?: string;
```
Comment (KATT) [inline components/dialog/ConfirmationDialogContent.tsx:12]: - use literal strings :)
- call it `variety` similar to material UI's [alert](https://material-ui.com/components/alert/)

```suggestion
  /**
   * @default info
   */
  variety?: 'error' | 'info' /* ... */;
```

**Review by KATT — CHANGES_REQUESTED** [review]

File: components/dialog/ConfirmationDialogContent.tsx [hunk 3]
```diff
@@ -31,11 +32,13 @@ export default function ConfirmationDialogContent({
           <DialogPrimitive.Description className="text-neutral-500">{children}</DialogPrimitive.Description>
         </div>
       </div>
-      <div className="mt-5 sm:mt-4 sm:flex sm:flex-row-reverse">
-        <DialogClose onClick={onConfirm} className="btn btn-primary">
+      <div className="mt-5 sm:mt-4 sm:flex sm:flex-row-reverse gap-x-2">
+        <DialogClose as={Button} color="primary" onClick={onConfirm}>
           {confirmBtnText}
```
Comment (KATT) [inline components/dialog/ConfirmationDialogContent.tsx:37]: if you want them optional
```suggestion
          {confirmBtnText ?? 'Confirm'}
```

File: components/dialog/ConfirmationDialogContent.tsx (hunk 1 above)
Comment (emrysal) [reply components/dialog/ConfirmationDialogContent.tsx:9]: image

Lint removed my newlines so it became less clear :(

Comment (KATT) [reply components/dialog/ConfirmationDialogContent.tsx:9]: Aaahhh, ignore then!

**Review by pumfleet — APPROVED** [review]

**Review by KATT — APPROVED** [review]

**Pull request merged**